draft-ietf-httpbis-retrofit.md

```diff
@@ -70,7 +70,22 @@ This specification defines how a selection of existing HTTP fields can be handle
 
 It does so using two techniques. {{compatible}} lists compatible fields -- those that can be handled as if they were Structured Fields due to the similarity of their defined syntax to that in Structured Fields. {{mapped}} lists mapped fields -- those whose syntax needs to be transformed into an underlying data model which is then mapped into that defined by Structured Fields.
 
-Note that while implementations can parse and serialise compatible fields as Structured Fields subject to the caveats in {{compatible}}, a sender cannot generate mapped fields from {{mapped}} and expect them to be understood and acted upon by the recipient without prior negotiation. This specification does not define such a mechanism.
+Note that this specification does not enable use of Retrofit Structured Fields in the HTTP protocol "on the wire" or in APIs; it only establishes handling for specific fields that might be used by such applications in the future.
+
+
+## Using Retrofit Structured Fields
+
+Retrofitting data structures onto existing and widely-deployed HTTP fields requires careful handling to assure interoperability and security. This section highlights considerations for applications that use Retrofit Structured Fields.
+
+While the majority of field values seen in HTTP traffic should be able to be parsed or mapped successfully, some will not. An application using Retrofit Structured Fields will need to define how unsuccessful values will be handled.
+
+For example, an API that exposes field values using Structured Fields data types might make the field value available as a string in cases where the field did not successfully parse or map.
```

Comment:
Three observations:

1. As a string or as a sequence of bytes?

2. This example really only applies to "Compatible" fields (which I think isn't quite the right name for these, perhaps "reinterpreted") not "mapped" fields.

3. The security benefits that accrue to natively-defined SF fields do not apply to compatible fields.  You are suggesting that you might build code like this:

```
if x is sf-type-that-is-expected:
    simple code
else if x is sf-*:
    drop x
else: # x wasn't retrofit successfully
    parse and validate x using more complex techniques
```

The simple code is the fast path here.  In most cases, code will hit the simple path and benefit from SF parsing, validation, and simpler processing.  But that final else clause remains a liability.  Despite it being close to never used, they are fully exposed to any vulnerabilities in that code.

Reply:
It's an example, not a directive. The point about security is interesting, and merits some text in SC; adding #2528.